eth_tester/utils/backend_testing.py

```diff
@@ -474,12 +486,112 @@ def test_estimate_gas(self, eth_tester):
             math_address,
             'increment',
         )
-        gas_estimation = eth_tester.estimate_gas(dissoc(estimate_call_math_transaction, 'gas'))
+        if not is_pyevm_available():  # py-evm requires the gas parameter
+            estimate_call_math_transaction = dissoc(estimate_call_math_transaction, 'gas')
+        gas_estimation = eth_tester.estimate_gas(estimate_call_math_transaction)
         call_math_transaction = assoc(estimate_call_math_transaction, 'gas', gas_estimation)
         transaction_hash = eth_tester.send_transaction(call_math_transaction)
         receipt = eth_tester.get_transaction_receipt(transaction_hash)
         assert receipt['gas_used'] == gas_estimation
 
+    def test_can_call_after_exception_raised_calling_pyethereum(self, eth_tester):
+        self.skip_if_no_evm_execution()
+        if is_pyevm_available():
+            pytest.skip('Test only relevant for pyethereum.')
+        elif is_pyethereum16_available():
+            from ethereum.tester import TransactionFailed
+        elif is_pyethereum21_available():
+            from ethereum.tools.tester import TransactionFailed
```

Comment:
Here's a good example of something that should be unified. These should all be the same exception, not dependent on the back end.

Wait until #43 is merged (just waiting on what I expect to be a green test), then use the same pattern from it to convert internal `TransactionFailed` exceptions into a standard `eth_tester.exceptions.TransactionFailed` exception.

Reply:
Ok, #43 is merged. You can rebase or merge master to get the convenience decorator from that PR.